src/components/PupilComponents/QuizMatchAnswer/QuizMatchAnswer.tsx

```diff
@@ -0,0 +1,145 @@
+  OakBox,
+  OakDraggableFeedback,
+  OakDroppable,
+  OakQuizMatch,
+  OakQuizMatchProps,
+  OakUL,
+} from "@oaknational/oak-components";
+
+import { isMatchAnswer } from "../QuizUtils/answerTypeDiscriminators";
+import { getStemTextData } from "../QuizUtils/stemUtils";
+
+import { useQuizEngineContext } from "@/components/PupilComponents/QuizEngineProvider";
+import { invariant } from "@/components/PupilComponents/pupilUtils/invariant";
+
+const StyledUL = styled(OakUL)`
+  list-style: none;
+  padding-inline: 0;
+`;
+
+export const QuizMatchAnswer = () => {
+  const {
+    currentQuestionData,
+    questionState,
+    currentQuestionIndex,
+    updateQuestionMode,
+  } = useQuizEngineContext();
+  invariant(currentQuestionData, "currentQuestionData is not defined");
+  const questionUid = currentQuestionData.questionUid;
+  const feedback = questionState[currentQuestionIndex]?.feedback;
+
+  invariant(
+    currentQuestionData.answers && isMatchAnswer(currentQuestionData.answers),
+    `current '${questionUid}' is not a match question`,
+  );
+
+  const answers = Object.fromEntries(
+    currentQuestionData.answers.match.map(
+      ({ correct_choice, match_option }) => {
+        const match = getStemTextData(match_option);
+        const choice = getStemTextData(correct_choice);
+
+        invariant(match?.text, "match_text is missing");
+        invariant(choice?.text, "choice_text is missing");
+
+        return [match.text, choice.text];
+      },
+    ),
+  );
+  const matchItems = Object.keys(answers).map((label, index) => ({
+    id: index.toString(),
+    label,
+  }));
+  const choiceItems = Object.values(answers).map((label, index) => ({
+    id: index.toString(),
+    label,
+  }));
+  const [currentMatches, setCurrentMatches] = useState<{
+    [matchId: string]: string;
+  }>({});
+  const handleChange: OakQuizMatchProps["onChange"] = (matches) => {
+    // Update the question mode to input if all matches have been made
+    // to enable the question to be submitted
+    updateQuestionMode(
+      Object.keys(matches).length === matchItems.length ? "input" : "init",
+    );
+    setCurrentMatches(
+      Object.fromEntries(
+        Object.entries(matches).map(([matchId, item]) => [matchId, item.id]),
+      ),
+    );
+  };
+
+  if (feedback) {
+    invariant(isArray(feedback), "question feedback is not an array");
+
+    return (
+      <StyledUL>
+        {matchItems.map(({ id, label }, i) => {
+          const currentFeedback = feedback.at(i);
+          const choice = choiceItems.find(
+            (choice) => choice.id === currentMatches[i],
+          );
+          const correctChoice = choiceItems.at(i);
+          invariant(
+            currentFeedback,
+            `feedback is missing for match '${label}'`,
+          );
+          invariant(choice, `choice is missing for match '${label}'`);
+          invariant(
+            correctChoice,
+            `correctChoice is missing for match '${label}'`,
+          );
```

Comment:
nulls, nulls everywhere!

Reply:
our schema is in a sorry state, but it's not worth refactoring until we full retire legacy data and we switch to the new approach for materialized views

Reply:
I'm just screaming into the void, no action is needed 😆